test/Esprima.Tests/JavascriptTest.cs

```diff
@@ -0,0 +1,425 @@
+﻿using Esprima.Utils;
+using Xunit;
+
+namespace Esprima.Tests
+{
+    public class JavascriptTest
+    {
+        [Fact]
+        public void ToJavascriptTest1()
+        {
+            var parser = new JavaScriptParser(@"if (true) { p(); }
+switch(foo) {
+    case 'A':
+        p();
+        break;
+}
+switch(foo) {
+    default:
+        p();
+        break;
+}
+for (var a = []; ; ) { }
+for (var elem of list) { }
+");
+            var program = parser.ParseScript();
+            var code = ToJavascriptConverter.ToJavascript(program);
+
+            Assert.Equal("if(true){p();}switch(foo){case 'A':p();break;}switch(foo){default:p();break;}for(var a=[];;){}for(var elem of list){}", code);

```

Comment:
This is the only test that actually checks the rendered javascript. Ideally any javavascript in should be the same out, but if we can't keep the original text because we should at least export with a common indentation pattern, and be able to check the output is fine. Maybe a better test in this case is to take any javascript file (call it source1), parse it, render it (call it source2) and ensure that parsing and rendering source2 is the same as source2.

This can be done for all subsequent tests in this file.

Reply:
I can fix the tests...
I many used them to test parts wich did not work, I will create an really expected output.
Of cause we need indention on the converter (if beautification is enabled), but I had no time atm to work on this.

Reply:
Is it generating valid JavaScript?

```
_sb.Append("static style=css`" + css + "`;");
```

If so, could you update the parse tree before it's rendered to JS instead?
If not then we'll have to live with this class being public for now. Or you could do some InternalVisibleFrom trick.

Reply:
yes
i could not update the tree, cause everthing is readonly

Reply:
cause of that I wanted a real visitor pattern like the ExpressionVisitor in C#, wich could return modified Expressions